cpp/src/parquet/stream_reader_test.cc

```diff
@@ -261,7 +275,176 @@ TEST_F(TestStreamReader, ValueChecking) {
     ASSERT_DOUBLE_EQ(d, TestData::GetDouble(i));
   }
+  ASSERT_EQ(reader_.current_row(), TestData::num_rows);
+  ASSERT_EQ(i, TestData::num_rows);
+}
+
+TEST_F(TestStreamReader, SkipRows) {
+  bool b;
+  std::string s;
+  std::array<char, 4> char_array;
+  char c;
+  int8_t int8;
+  uint16_t uint16;
+  int32_t int32;
+  uint64_t uint64;
+  std::chrono::microseconds ts_us;
+  float f;
+  double d;
+  std::string str;
+
+  const int num_rows_to_read = 3;
+  const int num_rows_to_skip = 13;
+  int num_rows_read = 0;
+  int i = 0;
+
+  // Skipping zero rows is ok.
+  //
+  ASSERT_EQ(0, reader_.SkipRows(0));
+
+  ASSERT_EQ(false, reader_.eof());
+  ASSERT_EQ(0, reader_.current_row());
+  ASSERT_EQ(TestData::num_rows, reader_.num_rows());
+
+  while (!reader_.eof()) {
+    for (int j = 0; !reader_.eof() && (j < num_rows_to_read); ++i, ++j) {
+      reader_ >> b;
+      reader_ >> s;
+      reader_ >> c;
+      reader_ >> char_array;
+      reader_ >> int8;
+      reader_ >> uint16;
+
+      // Not allowed to skip row once reading columns has started.
+      ASSERT_THROW(reader_.SkipRows(1), ParquetException);
+
+      reader_ >> int32;
+      reader_ >> uint64;
+      reader_ >> ts_us;
+      reader_ >> f;
+      reader_ >> d;
+      reader_ >> EndRow;
+      num_rows_read += 1;
+
+      ASSERT_EQ(b, TestData::GetBool(i));
+      ASSERT_EQ(s, TestData::GetString(i));
+      ASSERT_EQ(c, TestData::GetChar(i));
+      ASSERT_EQ(char_array, TestData::GetCharArray(i));
+      ASSERT_EQ(int8, TestData::GetInt8(i));
+      ASSERT_EQ(uint16, TestData::GetUInt16(i));
+      ASSERT_EQ(int32, TestData::GetInt32(i));
+      ASSERT_EQ(uint64, TestData::GetUInt64(i));
+      ASSERT_EQ(ts_us, TestData::GetChronoMicroseconds(i));
+      ASSERT_FLOAT_EQ(f, TestData::GetFloat(i));
+      ASSERT_DOUBLE_EQ(d, TestData::GetDouble(i));
+    }
+    ASSERT_EQ(num_rows_to_skip, reader_.SkipRows(num_rows_to_skip));
+    i += num_rows_to_skip;
+  }
+  ASSERT_EQ(TestData::num_rows, reader_.current_row());
+
+  ASSERT_EQ(num_rows_read, num_rows_to_read * TestData::num_rows /
+                               (num_rows_to_read + num_rows_to_skip));
+
+  // Skipping rows at eof is allowed.
+  //
+  ASSERT_EQ(0, reader_.SkipRows(100));
+}
+
+TEST_F(TestStreamReader, SkipAllRows) {
+  ASSERT_EQ(false, reader_.eof());
+  ASSERT_EQ(0, reader_.current_row());
+
+  ASSERT_EQ(reader_.num_rows(), reader_.SkipRows(2 * reader_.num_rows()));
+
+  ASSERT_EQ(true, reader_.eof());
+  ASSERT_EQ(reader_.num_rows(), reader_.current_row());
+}
+
+TEST_F(TestStreamReader, SkipColumns) {
+  bool b;
+  std::string s;
+  std::array<char, 4> char_array;
+  char c;
+  int8_t int8;
+  uint16_t uint16;
+  int32_t int32;
+  uint64_t uint64;
+  std::chrono::microseconds ts_us;
+  float f;
+  double d;
+  std::string str;
+
+  int i;
+
+  for (i = 0; !reader_.eof(); ++i) {
+    ASSERT_EQ(i, reader_.current_row());
+    ASSERT_EQ(0, reader_.current_column());
+
+    if (i % 31 == 0) {
```

Comment:
this test is particularly hard to read and it isn't clear where the different numbers come from.

Reply:
These numbers are arbitrary, just for testing purposes.
I have added comments to make this more clear.